src/main/telemetry/crsf.c

```diff
@@ -325,6 +361,42 @@ void crsfFrameDeviceInfo(sbuf_t *dst) {
     *lengthPtr = sbufPtr(dst) - lengthPtr;
 }
 
+
+#if defined(USE_CRSF_V3)
+void crsfFrameSpeedNegotiationResponse(sbuf_t *dst, bool reply) {
+
+    uint8_t *lengthPtr = sbufPtr(dst);
+    sbufWriteU8(dst, 0);
+    sbufWriteU8(dst, CRSF_FRAMETYPE_COMMAND);
+    sbufWriteU8(dst, CRSF_ADDRESS_CRSF_RECEIVER);
+    sbufWriteU8(dst, CRSF_ADDRESS_FLIGHT_CONTROLLER);
+    sbufWriteU8(dst, CRSF_COMMAND_SUBCMD_GENERAL);
+    sbufWriteU8(dst, CRSF_COMMAND_SUBCMD_GENERAL_CRSF_SPEED_RESPONSE);
+    sbufWriteU8(dst, crsfSpeed.portID);
+    sbufWriteU8(dst, reply);
+    crc8_poly_0xba_sbuf_append(dst, &lengthPtr[1]);
+    *lengthPtr = sbufPtr(dst) - lengthPtr;
+}
+
+static void crsfProcessSpeedNegotiationCmd(uint8_t *frameStart) {
+
+    uint32_t newBaudrate = frameStart[2] << 24 | frameStart[3] << 16 | frameStart[4] << 8 | frameStart[5];
+    uint8_t ii = 0;
+    for (ii = 0; ii < BAUD_COUNT; ++ii) {
```

Comment:
There is a `BAUD_RATE_COUNT` which should be used here instead of the enumeration to avoid out of bounds access...